Add YAKD - Kubernetes Dashboard addon

manusa opened the pull request:
This PR adds an addon for automatically setting up [YAKD (Yet Another Kubernetes Dashboard)](https://github.com/manusa/yakd).

This is an alternative Kubernetes Dashboard with special support for Minikube, OpenShift, and other flavors of Kubernetes.

It was originally started as an example project for [YAKC (Yet Another Kubernetes Client)](https://github.com/manusa/yakc) but it was recently spun off as a separate project.

Regarding Minikube, it offers specific functionality such as cluster detection, support for opening `NodePort` services from the interface, and more.

<!-- 🎉 Thank you for contributing to minikube! 🎉 Here are some hints to get your PR merged faster:

1. Your PR title will be included in the release notes, choose it carefully
2. If the PR fixes an issue, add "fixes #<issue number>" to the description.
3. If the PR is a user interface change, please include a "before" and "after" example.
4. If the PR is a large design change, please include an enhancement proposal:
https://github.com/kubernetes/minikube/tree/master/enhancements
-->

spowelljr commented on pkg/addons/addons.go:
```diff
@@ -150,6 +150,12 @@ kubectl get secret $SECRET --namespace headlamp --template=\{\{.data.token\}\} |
+	case "yakd":
+		out.Styled(style.Tip, `To access YAKD - Kubernetes Dashboard, wait for Pod to be ready and run the following command:
+
+	minikube service yakd-dashboard -n yakd-dashboard
```
```suggestion
	minikube{{.profileArg}} service yakd-dashboard -n yakd-dashboard
```

I'd recommend putting `{{.profileArg}}` there because if the user enables the addon on a cluster with a custom profile your current command will take place against the default profile.

Example:
```
$ minikube start -p p2 --addons yakd
...
To access YAKD - Kubernetes Dashboard, wait for Pod to be ready and run the following command:

minikube service yakd-dashboard -n yakd-dashboard

$ minikube service yakd-dashboard -n yakd-dashboard
🤷  Profile "minikube" not found. Run "minikube profile list" to view all profiles.
👉  To start a cluster, run: "minikube start"
```

Compared to:
```
$ minikube start -p p2 --addons yakd
...
To access YAKD - Kubernetes Dashboard, wait for Pod to be ready and run the following command:

minikube -p p2 service yakd-dashboard -n yakd-dashboard
```

spowelljr commented on site/content/en/docs/handbook/addons/yakd-kubernetes-dashboard.md:
```diff
@@ -0,0 +1,42 @@
+
+## YAKD - Kubernetes Dashboard Addon
+
+[YAKD - Kubernetes Dashboard](https://github.com/manusa/yakd) is a full-featured web-based Kubernetes Dashboard with special functionality for Minikube.
```
```suggestion
[YAKD - Kubernetes Dashboard](https://github.com/manusa/yakd) is a full-featured web-based Kubernetes Dashboard with special functionality for minikube.
```

spowelljr commented on test/integration/addons_test.go:
```diff
@@ -955,3 +956,16 @@ func validateNvidiaDevicePlugin(ctx context.Context, t *testing.T, profile strin
+func validateYakdAddon(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), "addons", "enable", "yakd", "-p", profile, "--alsologtostderr", "-v=1"))
+	if err != nil {
+		t.Fatalf("failed to enable yakd addon: args: %q: %v", rr.Command(), err)
+	}
```
Can you remove this command and just add `--addons=yakd` to line 107

spowelljr reviewed (changes requested):
It seems you're only building your image for amd64, would it be possible to also build it for arm64?

manusa commented:
Hey Steven,
Thanks for the review.
I've implemented the changes you requested.
I have also updated the image reference to a newer version that has images.
Let me know if there's anything else you need or if I should squash the commits.

spowelljr commented:
/ok-to-test

spowelljr commented on site/content/en/docs/handbook/addons/yakd-kubernetes-dashboard.md:
````diff
@@ -0,0 +1,42 @@
+minikube service yakd-dashboard -n yakd-dashboard
+```
+
+There dashboard will open in a new browser window and you should be able to start using it with no further hassle.
````
```suggestion
The dashboard will open in a new browser window and you should be able to start using it with no further hassle.
```

manusa replied:
Awesome, thx!

spowelljr reviewed (changes requested):
Other then typo this it looks good

spowelljr reviewed (approved):
Thanks for the PR!
